v2/blockstore/readonly.go

```diff
@@ -84,7 +85,9 @@ func (b *ReadOnly) DeleteBlock(_ cid.Cid) error {
 // Has indicates if the store contains a block that corresponds to the given key.
 func (b *ReadOnly) Has(key cid.Cid) (bool, error) {
 	offset, err := b.idx.Get(key)
-	if err != nil {
+	if errors.Is(err, index.ErrNotFound) {
+		return false, nil
+	} else if err != nil {
 		return false, err
 	}
```

Comment:
So, what you are saying is that if the Index dosen't have a cid, the blockstore dosen't have the cid but there's no error.

The other code path is that "something went wrong" when trying to lookup the Index.

Reply:
Why was this resulting into "not found" when building the UnixFS DAG ?

Reply:
This is the error coming from the index. The index only has a "get offset" method, which returns a non-nil error if a CID is not in the index, or if something went truly wrong - such as a corrupted index. In the former case, where everything went fine but a CID is simply missing, we get an ErrNotFound. In this non-fatal case, the blockstore interface wants Has to return a nil error, and to simply report that with a "false" return value.

> Why was this resulting into "not found" when building the UnixFS DAG ?

I added some debug prints to the blockstore. It seems like ipld-prime's "async commit" to the blockstore did a bunch of bs.Has calls before bs.Put, perhaps as a means to avoid duplicating work. Those were returning a seemingly fatal error of "not found", so the whole thing failed immediately.